plugins/package-managers/swiftpm/src/main/kotlin/SwiftPm.kt

```diff
@@ -80,7 +80,9 @@ class SwiftPm(
     }
 
     override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
-        requireLockfile(definitionFile.parentFile) { definitionFile.name != PACKAGE_SWIFT_NAME }
+        if (definitionFile.name == PACKAGE_SWIFT_NAME) {
```

Comment:
I don't understand why this is necessary. If the `definitionFile.name` is `PACKAGE_RESOLVED_NAME`, then `definitionFile.resolveSibling(PACKAGE_RESOLVED_NAME)` points to itself and the `isFile` check succeeds, and no `require()` check is triggered.

As a side node, IMO the name `requireLockfile()` is misleading / wrong, as the function does not require a lockfile to be present, but only `allowDynamicVersions || condition()` to be true. The function should probably be renamed, but I don't have a good proposal right now to what.

Reply:
> I don't understand why this is necessary. If the `definitionFile.name` is `PACKAGE_RESOLVED_NAME`, then `definitionFile.resolveSibling(PACKAGE_RESOLVED_NAME)` points to itself and the `isFile` check succeeds, and no `require()` check is triggered.

That's correct behavior then, because `Package.resolved` is the lockfile.

Reply:
> That's correct behavior then

I'm not trying to say the new code introduces incorrect behavior, but I was trying to explain why I believe the introduced check to be superfluous. So, do you agree the added check is not strictly necessary? That's still not clear to me.

Reply:
> I'm not trying to say that the new code introduces incorrect behavior, but I was trying to explain why I believe the introduced check to be superfluous.

I need to check this by running ORT. From reading the code, I assumed that when both files are present, the SwiftPm.kt would run the analysis twice: once for the code path for the lockfile and once based on `swift package show-dependencies`. (If you already know why this isn't the case, then let me know)

Reply:
You were right. I've dropped the commit. Good catch!

Reply:
Hmm, it now seems to me that you've dropped the wrong commit? This discussion belongs to the commit that adds the `if (definitionFile.name == PACKAGE_SWIFT_NAME)` check. You should be able to drop that as well.

Reply:
Hm. If we have the situation that `Package.swift` exists, and `allowDynamicVersions=false`, then we must do a check whether the `lockfile` exists, and fail in case it doesn't. This is the reason for the existance of the check. So, why can it be removed?

Reply:
> If the definition file is `Package.resolved` then of course no check is necessary, because we know that a lockfile exists as the definition file is a lockfile.

Right, the check is not *necessary* in that case, but it also does not hurt to have it.

> So, why is the check entirely unnecessary in your view?

Like I tried to explain [above](https://github.com/oss-review-toolkit/ort/pull/8154#discussion_r1462040805) it's simply not strictly required. It's maybe an optimization, but at the cost of added complexity and reduced readability. The original code would have worked just as well.

Maybe we should have a Huddle to do through this PR?